File: Python/nth-digit.py
class Solution:
    def findNthDigit(self, n):
        """
        :type n: int
        :rtype: int
        """
        length = 1
        x = 9

        while n > length * x:
            n -= length * x
            length += 1
            x *= 10

        indexAtNums = (n - 1) // length
        indexAtNum = (n - 1) % length
        num = 10 ** (length - 1 ) + indexAtNums
        res = num // (10 ** (length - 1 - indexAtNum))
        res %= 10

        return res

File: Swift/nth-digit.swift
class Solution {
    func findNthDigit(_ n: Int) -> Int {
        var n = n
        var len = 1
        var x = 9

        while n > len * x {
            n -= len * x
            len += 1
            x *= 10
        }

        let indexAtNums = (n - 1) / len
        let indexAtNum = (n - 1) % len
        let num = Int(pow(10, Double(len) - 1)) + indexAtNums
        var res = num / Int(pow(10, Double(len - 1 - indexAtNum)))
        res %= 10

        return res
    }
}
